### PR TITLE
Validate inputs

### PR DESCRIPTION
**Pull request by j-or**
Validates numeric and score inputs in data point inputs. 

Edit: Now changed to filter on isNumber instead of making values optional

### CODE REVIEW

File: packages/ui/src/features/data-points/useDatapoints.ts [hunk 1]
```diff
@@ -140,7 +135,10 @@ const _addRow = (original: DataEntry[], newRow: DataEntry) =>
       data: newRow.data.filter(dp => dp.value !== undefined),
       meta: {
         ...newRow.meta,
-        id: original.reduce((id, curr) => Math.max(id + 1, curr.meta.id), 1),
+        id:
```
Comment (j-or) [inline packages/ui/src/features/data-points/useDatapoints.ts:138]: Fixes a bug where data points would get the same id if data points had been deleted from the table.

File: packages/ui/src/features/core/info-box/info-box.style.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ export const useStyles = makeStyles()(theme => ({
     alignItems: 'center',
     fontSize: 14,
     borderRadius: '4px',
+    color: 'black',
```
Comment (j-or) [inline packages/ui/src/features/core/info-box/info-box.style.ts:9]: Fixes InfoBox showing theme coloured text when imported in other projects

File: packages/ui/src/features/data-points/useDatapoints.ts [hunk 3]
```diff
@@ -89,6 +90,11 @@ const convertToDataEntry = (
 ) => {
   const data = row.dataPoints
     .filter(d => d.value !== undefined)
+    .filter(
```
Comment (langdal) [inline packages/ui/src/features/data-points/useDatapoints.ts:93]: This filter removes all categorical values :)

File: packages/ui/src/features/data-points/useDatapoints.ts [hunk 4]
```diff
@@ -89,6 +90,11 @@ const convertToDataEntry = (
...
+      d =>
+        !categoricalVariables.map(v => v.name).includes(d.name) &&
+        isNumber(d.value)
```
Comment (langdal) [inline packages/ui/src/features/data-points/useDatapoints.ts:96]:
```suggestion
        categoricalVariables.map(v => v.name).includes(d.name) || isNumber(d.value)
```

**Review by langdal — APPROVED** [review]
ok when the filter is fixed

**Pull request merged**